lxd/patches.go

```diff
@@ -1876,3 +1877,60 @@ func patchStorageApiUpdateStorageConfigs(name string, d *Daemon) error {
 
 	return nil
 }
+
+func patchStorageApiLxdOnBtrfs(name string, d *Daemon) error {
+	pools, err := dbStoragePools(d.db)
+	if err != nil {
+		if err == NoSuchObjectError {
+			return nil
+		}
+		shared.LogErrorf("Failed to query database: %s", err)
+		return err
+	}
+
+	for _, poolName := range pools {
+		_, pool, err := dbStoragePoolGet(d.db, poolName)
+		if err != nil {
+			shared.LogErrorf("Failed to query database: %s", err)
+			return err
+		}
+
+		// Make sure that config is not empty.
+		if pool.Config == nil {
+			pool.Config = map[string]string{}
+
+			// Insert default values.
+			err = storagePoolFillDefault(poolName, pool.Driver, pool.Config)
+			if err != nil {
+				return err
+			}
+		}
+
+		if d.BackingFs != "btrfs" {
+			continue
+		}
+
+		if pool.Driver != "btrfs" {
+			continue
+		}
+
+		source := pool.Config["source"]
+		cleanSource := filepath.Clean(source)
+		loopFilePath := shared.VarPath("disks", poolName+".img")
+		if cleanSource != loopFilePath {
+			continue
+		}
```

Comment:
Hmm, what if I have /var/lib/lxd on btrfs and did:

    lxc storage create blah btrfs

Which gets me a pool with source=/var/lib/lxd/disks/blah.img, won't that code now change it to source=/var/lib/lxd/storage-pools/blah and break it?